prediction_market_agent/agents/autogen_general_agent/social_agent.py

```diff
@@ -0,0 +1,200 @@
+
+import autogen
+from autogen import AssistantAgent, UserProxyAgent
+from autogen.cache import Cache
+from loguru import logger
+from prediction_market_agent_tooling.markets.omen.data_models import OmenBet
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
+    OmenSubgraphHandler,
+)
+from prediction_market_agent_tooling.tools.utils import utcnow
+from pydantic import BaseModel
+from web3 import Web3
+
+from prediction_market_agent.agents.autogen_general_agent.prompts import (
+    CRITIC_PROMPT,
+    INFLUENCER_PROMPT,
+)
+from prediction_market_agent.utils import APIKeys
+
+
+# We have an influencer and a critic. Based on https://microsoft.github.io/autogen/docs/notebooks/agentchat_nestedchat/.
+
+
+# Added this class because properties are not automatically dumped when doing class.dict(include=...),
+# hence new class with less fields is the easiest approach.
+class BetInputPrompt(BaseModel):
+    title: str
+    boolean_outcome: bool
+    collateralAmountUSD: float
+    creation_datetime: datetime
+    probability: float
+
+
+class AutogenAgentType(str, Enum):
+    WRITER = "writer"
+    CRITIC = "critic"
+    USER = "user"
+
+
+def reflection_message(
+    recipient: UserProxyAgent,
+    messages: list[Dict[Any, Any]],
+    sender: AssistantAgent,
+    config: Optional[Any] = None,
+) -> str:
+    reflect_prompt = Template(CRITIC_PROMPT).substitute(
+        TWEET=recipient.chat_messages_for_summary(sender)[-1]["content"]
+    )
+    return reflect_prompt
+
+
+def build_llm_config(model: str) -> Dict[str, Any]:
+    keys = APIKeys()
+    return {
+        "config_list": [
+            {"model": model, "api_key": keys.openai_api_key.get_secret_value()}
+        ],
+    }
+
+
+def build_agents(model: str) -> Dict[AutogenAgentType, autogen.ConversableAgent]:
+    llm_config = build_llm_config(model)
+
+    user_proxy = autogen.UserProxyAgent(
+        name="User",
+        human_input_mode="NEVER",
+        is_termination_msg=lambda x: x.get("content", "").find("TERMINATE") >= 0,
+        code_execution_config={
+            "last_n_messages": 1,
+            "work_dir": "tasks",
+            "use_docker": False,
+        },
+    )
+
+    writer = autogen.AssistantAgent(
+        name="Writer",
+        llm_config=llm_config,
+        system_message="""
+        You are a professional influencer, known for your insightful and engaging tweets.
+        You transform complex concepts into compelling narratives.
+        You should imporve the quality of the content based on the feedback from the user.
+        You must always return only the tweet.
+        """,
+    )
+
+    critic = autogen.AssistantAgent(
+        name="Critic",
+        llm_config=llm_config,
+        system_message="""
+            You are a critic, known for your thoroughness and commitment to standards.
+            Your task is to scrutinize content for any harmful elements or regulatory violations, ensuring
```

Comment:
Is this the part of the prompt that causes the critic to censor the talk of 'gambling'?

Reply:
Yes, I rephrased that and now it is less restrictive, since we are at the of the day indeed betting.